b2share/modules/files/tasks.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+# Copyright (C) 2017 CERN.
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Celery background tasks."""
+import json
+from celery import shared_task
+from flask import current_app, url_for
+from flask_babelex import lazy_gettext as _
+from invenio_db import db
+from invenio_records_files.models import RecordsBuckets
+from invenio_files_rest.models import FileInstance, ObjectVersion
+from invenio_files_rest.tasks import schedule_checksum_verification
+from invenio_mail.tasks import send_email
+
+
+def failed_checksum_files_query():
+    return FileInstance.query.filter_by(last_check=None)
+
+
+def notify_admin():
+    results = db.session.query(RecordsBuckets.record_id,
+                               ObjectVersion.bucket_id,
+                               ObjectVersion.key,
+                               FileInstance.uri,
+                               FileInstance.checksum).\
+        filter(ObjectVersion.file_id == FileInstance.id,
+               FileInstance.last_check == False).all()
+    corrupted_files = {}
+    for index, result in enumerate(results):
+        corrupted_files[index] = \
+            'Link: u"<a href=' + current_app.config['SERVER_NAME'] + \
+            url_for('b2share_records_rest.b2rec_item',
+                    pid_value=str(result[0])) + \
+            '>' + str(result[0]) + '"</a>"' + \
+            '\n Record Id: {0}\n Bucket Id: {1}\n File key: {2}\n' + \
+            ' Path: {3}\n Old Checksum: {4}\n\n'.\
+            format(str(result[0]), str(result[1]),
+                   str(result[2]), str(result[3]),
+                   str(result[4]))
+    error_count = FileInstance.query.filter_by(last_check=None).count()
+
+    msg_content = '{0}: {1}\n{2}: {3}'.\
+        format(_('Possibly Corrupted Files'),
```

Comment:
The files are not corrupted at that point. It is more probable that the filesystem is not reachable. We can just report "An error occurred while checking files checksum".